pkg/ovsdb/database.go

```diff
@@ -150,32 +159,64 @@ func (con *DatabaseEtcd) PutData(ctx context.Context, key string, obj interface{
 	}
 	return nil
 }
-func (con *DatabaseEtcd) AddMonitor(keysPrefix string, mcr ovsjson.MonitorCondRequest, isV1 bool, hand *handlerKey) {
+
+func (con *DatabaseEtcd) AddMonitors(dbName string, updaters map[string][]*updater, handler handlerKey)  {
+	fmt.Println("DB Monitor 1")
 	con.mu.Lock()
 	defer con.mu.Unlock()
-	if monitor, ok := con.monitors[keysPrefix]; !ok {
-		con.monitors[keysPrefix] = *newMonitor(con.cli, con.prefix+keysPrefix, mcr, isV1, hand)
-	} else {
-		monitor.addHandler(mcr, isV1, hand)
+	fmt.Println("DB Monitor 2")
+	m, ok := con.monitors[dbName]
+	fmt.Printf("DB Monitor 3 %v\n", ok)
+	if !ok {
+		// we don't have monitors for this database yet, create a new one.
+		m = newMonitor(dbName, con)
+		ctxt, cancel := context.WithCancel(context.Background())
+		m.cancel = cancel
+		wch := con.cli.Watch(clientv3.WithRequireLeader(ctxt),con.prefix + dbName,
+			clientv3.WithPrefix(),
+			clientv3.WithCreatedNotify(),
+			clientv3.WithPrevKV())
+		m.watchChannel = wch
+		con.monitors[dbName] = m
+	}
+	fmt.Println("DB Monitor 4")
+	m.addUpdaters(con.prefix, updaters, handler)
+	fmt.Println("DB Monitor 5")
+	if !ok {
+		m.start()
 	}
+	fmt.Println("DB Monitor 6")
```

Comment:
maybe you want to remove some of the prints ...

Reply:
yeap

Reply:
done